error on trying to cd to directory without executable bit

adqm opened the pull request:
This adds a simple check to report an error if a user tries to `cd` into a directory they don't have permissions to enter, per #557.

scopatz commented on xonsh/dirstack.py:
```diff
@@ -88,6 +88,8 @@ def cd(args, stdin=None):
     if not os.path.isdir(d):
         return '', 'cd: {0} is not a directory\n'.format(d)
+    if not os.access(d, os.X_OK):
+        return '', 'cd: permission denied: {0}\n'.format(d)
```
Just an idea, but now we have the option of returning a returncode as the third element of the tuple. Maybe you could add a `, 1` to the end of this line?

adqm commented:
Alright, I went in and added return codes to `cd`, as well as the other commands in `dirstack.py`.

Also fixed a bug with `dirs -c` (must not be a very widely-used command since it made it this long without being noticed :) )

scopatz commented:
Thanks a ton @adqm!  And yeah, I never use dirstack, but I am glad that it is around :)
